### DISCUSSION TITLE
How to create custom action natively with python?

### DISCUSSION THREAD

**Discussion by dehamzah** [cloud-custodian/cloud-custodian#7550] [General]
Hi, I did check the cloud custodian docs have a variety of built-in actions, but there are things that not fit my use case and I want to create the custom action. I see the docs is suggesting to use lambda function if I use AWS, but that makes me to setup permission also with the lambda function. Is there possible to create custom action using python script and use it in the policy? If its possible can you guide me with the correct way of doing it?

**Comment by thisisshi** [comment]
The only way to trigger a custom piece of code would be through the invoke-lambda action. If you want to contribute an action feel free to open a pull request if it's applicable to the community
